### PR TITLE
SONAR-8173 drop global permission "shareDashboard"

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/version/v62/DeletePermissionShareDashboard.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    super(db);
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    deleteFromTable(context, "group_roles");
+    deleteFromTable(context, "user_roles");
+  }
+
+  private void deleteFromTable(Context context, String tableName) throws SQLException {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/version/v62/DeletePermissionShareDashboard.java]: MAJOR Make "deleteFromTable" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [v62](https://github.com/SonarSource/sonarqube/blob/19033a897ccbf7e46a3ba391ade7ad3e2d8aaff4/sonar-db/src/main/java/org/sonar/db/version/v62): Add a 'package-info.java' file to document the 'v62' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

File: sonar-db/src/main/java/org/sonar/db/version/v62/DeletePermissionShareDashboard.java (hunk 1 above)
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/version/v62/DeletePermissionShareDashboard.java]: Seb probably added it in the PR that adds migrations 1400 and 1401.

**Comment by simonbrandhof** [comment]
Seb probably added missing package_info.java in the PR that adds migrations 1400 and 1401.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [v62](https://github.com/SonarSource/sonarqube/blob/488a1e7443d722fd6eff2321b85051361fc6bce0/sonar-db/src/main/java/org/sonar/db/version/v62): Add a 'package-info.java' file to document the 'v62' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

File: sonar-db/src/test/java/org/sonar/db/version/v62/DeletePermissionShareDashboardTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.sql.SQLException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.api.utils.System2;
+import org.sonar.db.DbTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DeletePermissionShareDashboardTest {
```
Comment (sns-seb) [inline sonar-db/src/test/java/org/sonar/db/version/v62/DeletePermissionShareDashboardTest.java]: missing test on delete from table USER_ROLES
some extra line returns to remove

File: sonar-db/src/test/java/org/sonar/db/version/v62/DeletePermissionShareDashboardTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  @Rule
+  public DbTester db = DbTester.createForSchema(System2.INSTANCE, DeletePermissionShareDashboardTest.class, "roles.sql");
+
+  private DeletePermissionShareDashboard underTest = new DeletePermissionShareDashboard(db.database());
+
+  @Test
+  public void delete_permissions() throws SQLException {
+    db.executeInsert("group_roles", "id", "1", "group_id", "1", "resource_id", "100", "role", "shareDashboard");
+    db.executeInsert("group_roles", "id", "2", "group_id", null, "resource_id", null, "role", "shareDashboard");
```
Comment (sns-seb) [inline sonar-db/src/test/java/org/sonar/db/version/v62/DeletePermissionShareDashboardTest.java]: missing group permission "shareDashboard" for anyone but specific to a project
missing group permission other than "shareDashboard" for a given group, a given group and project, anyone for a project (to ensure, only the right rows are deleted)

Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/version/v62/DeletePermissionShareDashboardTest.java]: `shareDashboard` is never defined on projects. Still this case can be covered.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request closed without merging**